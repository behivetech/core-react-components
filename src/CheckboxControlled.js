// Vendor Libs
import React, {Component} from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';

// Components
import Checkbox from './Checkbox';
import withFormControl from './withFormControl';
import withSwitchControl from './withSwitchControl';

class CheckboxControlled extends Component {
	render() {
		return (
			<Checkbox
				{...this.props}
				className={classnames('checkbox-controlled', this.props.className)}
			/>
		);
	}
}

CheckboxControlled.propTypes = {
    /** Ability to add additional className to the component for addistional styling from parent */
    className: PropTypes.string,
};

export default withFormControl(withSwitchControl(CheckboxControlled));
