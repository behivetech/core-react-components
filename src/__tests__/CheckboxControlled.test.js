import CheckboxControlled from '../CheckboxControlled';

jest.mock('../withFormControl');
jest.mock('../withSwitchControl');

describe('CheckboxControlled', () => {
    it('should render with the correct props', () => {
        const wrapper = shallow(<CheckboxControlled name="mockName" className="mock-class" />);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.dive()).toMatchSnapshot();
        expect(wrapper.dive().dive()).toMatchSnapshot();
    });
});