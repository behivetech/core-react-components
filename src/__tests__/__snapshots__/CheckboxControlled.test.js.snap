// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckboxControlled should render with the correct props 1`] = `
<SwitchControlled
  className="mock-class"
  formState={
    Object {
      "formDisabled": false,
      "getFieldError": [MockFunction],
      "getFieldValue": [MockFunction],
      "initialSubmit": false,
      "setFieldValidators": [MockFunction],
      "setFieldValue": [MockFunction],
      "setFieldValueDebounced": [MockFunction],
      "shouldDisable": [MockFunction],
    }
  }
  name="mockName"
/>
`;

exports[`CheckboxControlled should render with the correct props 2`] = `
<CheckboxControlled
  className="mock-class"
  name="mockName"
/>
`;

exports[`CheckboxControlled should render with the correct props 3`] = `
<Checkbox
  checked={false}
  className="checkbox-controlled mock-class"
  name="mockName"
  onChange={[Function]}
  value={true}
/>
`;
